### PR TITLE
Make sample analyzer more friendly for running locally.

### PR DESCRIPTION
**Pull request by gspencergoog**
Now the sample analyzer can be run locally with:
```bash
dart dev/bots/analyze-sample-code.dart --temp=/tmp/samples
```
And it leaves /tmp/samples around so you can take a look at the code that was generated to see if it's the artificial environment that the samples are evaluated in that is the culprit for a failed analysis.

Before, you had to specify the whole path to the dart executable, and had to modify the code to keep the tempdir around.

### CODE REVIEW

File: dev/bots/analyze-sample-code.dart [hunk 1]
```diff
@@ -128,10 +164,14 @@ class SampleChecker {
...
+  String get _dartExecutable {
+    final File dartExecutable = File(Platform.resolvedExecutable);
+    return dartExecutable.absolute.path;
+  }
+
   static List<File> _listDartFiles(Directory directory, {bool recursive = false}) {
-    return directory.listSync(recursive: recursive, followLinks: false)
-      .whereType<File>()
-      .where((File file) => path.extension(file.path) == '.dart').toList();
+    return directory.listSync(recursive: recursive, followLinks: false).whereType<File>().where((File file) => path.extension(file.path) == '.dart').toList();
```
Comment (goderbauer) [inline dev/bots/analyze-sample-code.dart]: nit: this line seemed easier to read with the line breaks.

File: dev/bots/analyze-sample-code.dart [hunk 2]
```diff
@@ -104,9 +135,14 @@ class SampleChecker {
...
 
+  /// Whether or not to keep the temp directory around after running.
+  ///
+  /// Defaults to false.
+  final bool _keepTmp;
+
   /// The temporary directory where all output is written. This will be deleted
   /// automatically if there are no errors.
-  Directory _tempDir;
+  Directory tempDirectory;
```
Comment (goderbauer) [inline dev/bots/analyze-sample-code.dart]: Why does this need to be public now?

**Review by goderbauer — APPROVED** [review]
LGTM

File: dev/bots/analyze-sample-code.dart (hunk 1 above)
Comment (gspencergoog) [reply dev/bots/analyze-sample-code.dart]: Yeah, that was an unintentional dartfmt change.

**Pull request merged**